### PR TITLE
style[next]: improve typing in next.type_translation

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

File: src/gt4py/next/type_system/type_translation.py [hunk 1]
```diff
@@ -188,18 +199,23 @@ def from_value(value: Any) -> ts.TypeSpec:
...
+        elems = [from_value(el) for el in value]
+        assert all(isinstance(elem, ts.DataType) for elem in elems)
+        return ts.TupleType(types=elems)  # type: ignore[arg-type] # checked in assert
     else:
         type_ = xtyping.infer_type(value, annotate_callable_kwargs=True)
         symbol_type = from_type_hint(type_)
 
-    if isinstance(symbol_type, (ts.DataType, ts.CallableType, ts.OffsetType, ts.DimensionType)):
+    if isinstance(
+        symbol_type, (ts.DataType, ts.CallableType, ts.OffsetType, ts.DimensionType)
```
Comment (havogt) [inline src/gt4py/next/type_system/type_translation.py:217]: CallableType is not a TypeSpec. this looks like a bug

Comment (tehrengruber) [reply src/gt4py/next/type_system/type_translation.py:217]: CallableType is a mixin and all subtypes of it are a TypeSpec. Maybe we should rename it to something like `CallableTypeMixin`. If you want it to be 100% correct maybe `isinstance(symbol_type, ts.CallableType) and isinstance(symbol_type, ts.TypeSpec)`?

Comment (havogt) [reply src/gt4py/next/type_system/type_translation.py]: fixed by `isinstance(symbol_type, ts.CallableType) and isinstance(symbol_type, ts.TypeSpec)`

**Review by tehrengruber — APPROVED** [review]

File: src/gt4py/next/type_system/type_translation.py [hunk 2]
```diff
@@ -66,7 +67,7 @@ def from_type_hint(
     localns: Optional[dict[str, Any]] = None,
 ) -> ts.TypeSpec:
     recursive_make_symbol = functools.partial(from_type_hint, globalns=globalns, localns=localns)
-    extra_args = ()
+    extra_args: list = []
```
[the remark this answers is not in the document]
Comment (havogt) [reply src/gt4py/next/type_system/type_translation.py:70]: it's filled from `typing.get_args` which returns `Any`, could add additional checks, but thought better than before.

**Pull request merged**